[jerryx/arg]add jerryx_arg_utf8_string

jiangzidong opened the pull request:
JerryScript-DCO-1.0-Signed-off-by: Zidong Jiang zidong.jiang@intel.com

jiangzidong commented:
related issue: #2132

martijnthe commented on docs/09.EXT-REFERENCE-ARG.md:
```diff
@@ -461,7 +462,7 @@ jerryx_arg_boolean (bool *dest,
 
 Create a validation/transformation step (`jerryx_arg_t`) that expects to
-consume one `string` JS argument and stores it into a C `char` array.
+consume one `string` JS argument and stores it into a cesu8 C `char` array.
```
"CESU-8"? I think that's how one would normally write it (outside of the context of code). For example https://en.wikipedia.org/wiki/CESU-8

martijnthe commented on docs/09.EXT-REFERENCE-ARG.md:
```diff
@@ -482,6 +483,36 @@ jerryx_arg_string (char *dest,
+**Summary**
+
+Create a validation/transformation step (`jerryx_arg_t`) that expects to
+consume one `string` JS argument and stores it into a utf8 C `char` array.
```
"UTF-8" instead of "utf8"

martijnthe commented on jerry-ext/arg/arg-transform-functions.c:
```diff
@@ -251,21 +251,37 @@ jerryx_arg_transform_boolean (jerryx_arg_js_iterator_t *js_arg_iter_p, /**< avai
 
 /**
  * The common routine for string transformer.
+ * It works for both cesu8 and utf8 string.
```
Same here?

martijnthe commented:
Couple of small documentation nits but lgtm otherwise, thanks for adding.

jiangzidong commented:
@akosthekiss @zherczeg Could you please help review this one? add a new type into the jerryx_arg. Thanks!

martijnthe commented:
@akosthekiss @zherczeg could you take a look again? Thanks much.

LaszloLango commented on jerry-ext/arg/arg-transform-functions.c:
```diff
@@ -319,12 +337,64 @@ jerryx_arg_transform_string (jerryx_arg_js_iterator_t *js_arg_iter_p, /**< avail
+} /* jerryx_arg_transform_string_common */
+
+/**
+ * Tranform a JS argument to a cesu8 char array. Type coercion is not allowed.
```
typo: `Transform...`

LaszloLango commented on jerry-ext/arg/arg-transform-functions.c:
```diff
@@ -319,12 +337,64 @@ jerryx_arg_transform_string (jerryx_arg_js_iterator_t *js_arg_iter_p, /**< avail
+/**
+ * Tranform a JS argument to a cesu8 char array. Type coercion is not allowed.
+ *
+ * @return jerry undefined: the transformer passes,
```
Please add a note to free the return value like in jerry.c:

```
 *
 * Note:
 *      returned value must be freed with jerry_release_value, when it is no longer needed.
 *
 * @return ...
```

LaszloLango commented on jerry-ext/arg/arg-transform-functions.c:
```diff
@@ -319,12 +337,64 @@ jerryx_arg_transform_string (jerryx_arg_js_iterator_t *js_arg_iter_p, /**< avail
+/**
+ * Tranform a JS argument to a utf8 char array. Type coercion is not allowed.
+ *
+ * @return jerry undefined: the transformer passes,
```
ditto

LaszloLango commented on jerry-ext/arg/arg-transform-functions.c:
```diff
@@ -319,12 +337,64 @@ jerryx_arg_transform_string (jerryx_arg_js_iterator_t *js_arg_iter_p, /**< avail
+/**
+ * Tranform a JS argument to a cesu8 char array. Type coercion is allowed.
+ *
+ * @return jerry undefined: the transformer passes,
```
ditto

LaszloLango commented on jerry-ext/arg/arg-transform-functions.c:
```diff
@@ -319,12 +337,64 @@ jerryx_arg_transform_string (jerryx_arg_js_iterator_t *js_arg_iter_p, /**< avail
+/**
+ * Tranform a JS argument to a utf8 char array. Type coercion is allowed.
+ *
+ * @return jerry undefined: the transformer passes,
```
ditto

LaszloLango commented on jerry-ext/arg/arg-transform-functions.c:
```diff
@@ -319,12 +337,64 @@ jerryx_arg_transform_string (jerryx_arg_js_iterator_t *js_arg_iter_p, /**< avail
+} /* jerryx_arg_transform_string_strict */
+
+/**
+ * Tranform a JS argument to a utf8 char array. Type coercion is not allowed.
```
typo: Transform...

LaszloLango commented on jerry-ext/arg/arg-transform-functions.c:
```diff
@@ -319,12 +337,64 @@ jerryx_arg_transform_string (jerryx_arg_js_iterator_t *js_arg_iter_p, /**< avail
+} /* jerryx_arg_transform_utf8_string_strict */
+
+/**
+ * Tranform a JS argument to a cesu8 char array. Type coercion is allowed.
```
typo: Transform...

LaszloLango commented on jerry-ext/arg/arg-transform-functions.c:
```diff
@@ -319,12 +337,64 @@ jerryx_arg_transform_string (jerryx_arg_js_iterator_t *js_arg_iter_p, /**< avail
 } /* jerryx_arg_transform_string */
 
+/**
+ * Tranform a JS argument to a utf8 char array. Type coercion is allowed.
```
typo: Transform...

LaszloLango commented on jerry-ext/include/jerryscript-ext/arg.impl.h:
```diff
@@ -242,6 +243,51 @@ jerryx_arg_string (char *dest, /**< pointer to the native char array where the r
+ * @return a jerryx_arg_t instance.
+ */
+static inline jerryx_arg_t
+jerryx_arg_utf8_string (char *dest, /**< pointer to the native char array where the result should be stored */
```
`/**< [out] pointer to the native char array where the result should be stored */`

jiangzidong commented:
updated @LaszloLango , Thanks!

jiangzidong commented:
Hmm, the CI error for Mbed/k64f
seems the network connection error. How can I re-run the CI manually? @LaszloLango 

```

Worker information
hostname: ab80a7f7-5ccc-48cd-ad00-49b22d79342d@1.production-3-worker-org-f-1-gce
version: v3.3.1 https://github.com/travis-ci/worker/tree/1153e5010ef9863bf96264b75fba506109633460
instance: travis-job-3132ee34-8996-4436-ab25-1b97c4f0caa0 travis-ci-garnet-trusty-1512502259-986baf0 (via amqp)
startup: 21.438093237s
system_info
Build system information
Build language: c
Build group: stable
Build dist: trusty
Build id: 315690207
Job id: 315690224
Runtime kernel version: 4.4.0-101-generic
travis-build version: 4d559ff16
Build image provisioning date and time
Tue Dec  5 19:58:13 UTC 2017
Operating System Details
Distributor ID:	Ubuntu
Description:	Ubuntu 14.04.5 LTS
Release:	14.04
Codename:	trusty
Cookbooks Version
7c2c6a6 https://github.com/travis-ci/travis-cookbooks/tree/7c2c6a6
git version
git version 2.15.1
bash version
GNU bash, version 4.3.11(1)-release (x86_64-pc-linux-gnu)
gcc version
gcc (Ubuntu 4.8.4-2ubuntu1~14.04.3) 4.8.4
Copyright (C) 2013 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
docker version
Client:
 Version:      17.09.0-ce
 API version:  1.32
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:42:38 2017
 OS/Arch:      linux/amd64
Server:
 Version:      17.09.0-ce
 API version:  1.32 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:41:20 2017
 OS/Arch:      linux/amd64
 Experimental: false
clang version
clang version 5.0.0 (tags/RELEASE_500/final)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /usr/local/clang-5.0.0/bin
jq version
jq-1.5
bats version
Bats 0.4.0
shellcheck version
0.4.6
shfmt version
v2.0.0
ccache version
ccache version 3.1.9
Copyright (C) 2002-2007 Andrew Tridgell
Copyright (C) 2009-2011 Joel Rosdahl
This program is free software; you can redistribute it and/or modify it under
the terms of the GNU General Public License as published by the Free Software
Foundation; either version 3 of the License, or (at your option) any later
version.
cmake version
cmake version 3.9.2
CMake suite maintained and supported by Kitware (kitware.com/cmake).
heroku version
heroku-cli/6.14.39-addc925 (linux-x64) node-v9.2.0
imagemagick version
Version: ImageMagick 6.7.7-10 2017-07-31 Q16 http://www.imagemagick.org
md5deep version
4.2
mercurial version
Mercurial Distributed SCM (version 4.2.2)
(see https://mercurial-scm.org for more information)
Copyright (C) 2005-2017 Matt Mackall and others
This is free software; see the source for copying conditions. There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
mysql version
mysql  Ver 14.14 Distrib 5.6.33, for debian-linux-gnu (x86_64) using  EditLine wrapper
openssl version
OpenSSL 1.0.1f 6 Jan 2014
packer version
Packer v1.0.2
Your version of Packer is out of date! The latest version
is 1.1.2. You can update by downloading from www.packer.io
postgresql client version
psql (PostgreSQL) 9.6.6
ragel version
Ragel State Machine Compiler version 6.8 Feb 2013
Copyright (c) 2001-2009 by Adrian Thurston
subversion version
svn, version 1.8.8 (r1568071)
   compiled Aug 10 2017, 17:20:39 on x86_64-pc-linux-gnu
Copyright (C) 2013 The Apache Software Foundation.
This software consists of contributions made by many people;
see the NOTICE file for more information.
Subversion is open source software, see http://subversion.apache.org/
The following repository access (RA) modules are available:
* ra_svn : Module for accessing a repository using the svn network protocol.
  - with Cyrus SASL authentication
  - handles 'svn' scheme
* ra_local : Module for accessing a repository on local disk.
  - handles 'file' scheme
* ra_serf : Module for accessing a repository via WebDAV protocol using serf.
  - using serf 1.3.3
  - handles 'http' scheme
  - handles 'https' scheme
sudo version
Sudo version 1.8.9p5
Configure options: --prefix=/usr -v --with-all-insults --with-pam --with-fqdn --with-logging=syslog --with-logfac=authpriv --with-env-editor --with-editor=/usr/bin/editor --with-timeout=15 --with-password-timeout=0 --with-passprompt=[sudo] password for %p:  --without-lecture --with-tty-tickets --disable-root-mailer --enable-admin-flag --with-sendmail=/usr/sbin/sendmail --with-timedir=/var/lib/sudo --mandir=/usr/share/man --libexecdir=/usr/lib/sudo --with-sssd --with-sssd-lib=/usr/lib/x86_64-linux-gnu --with-selinux
Sudoers policy plugin version 1.8.9p5
Sudoers file grammar version 43
Sudoers path: /etc/sudoers
Authentication methods: 'pam'
Syslog facility if syslog is being used for logging: authpriv
Syslog priority to use when user authenticates successfully: notice
Syslog priority to use when user authenticates unsuccessfully: alert
Send mail if the user is not in sudoers
Use a separate timestamp for each user/tty combo
Lecture user the first time they run sudo
Root may run sudo
Allow some information gathering to give useful error messages
Require fully-qualified hostnames in the sudoers file
Visudo will honor the EDITOR environment variable
Set the LOGNAME and USER environment variables
Length at which to wrap log file lines (0 for no wrap): 80
Authentication timestamp timeout: 15.0 minutes
Password prompt timeout: 0.0 minutes
Number of tries to enter a password: 3
Umask to use or 0777 to use user's: 022
Path to mail program: /usr/sbin/sendmail
Flags for mail program: -t
Address to send mail to: root
Subject line for mail messages: *** SECURITY information for %h ***
Incorrect password message: Sorry, try again.
Path to authentication timestamp dir: /var/lib/sudo
Default password prompt: [sudo] password for %p: 
Default user to run commands as: root
Value to override user's $PATH with: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin
Path to the editor for use by visudo: /usr/bin/editor
When to require a password for 'list' pseudocommand: any
When to require a password for 'verify' pseudocommand: all
File descriptors >= 3 will be closed before executing a command
Environment variables to check for sanity:
	TZ
	TERM
	LINGUAS
	LC_*
	LANGUAGE
	LANG
	COLORTERM
Environment variables to remove:
	RUBYOPT
	RUBYLIB
	PYTHONUSERBASE
	PYTHONINSPECT
	PYTHONPATH
	PYTHONHOME
	TMPPREFIX
	ZDOTDIR
	READNULLCMD
	NULLCMD
	FPATH
	PERL5DB
	PERL5OPT
	PERL5LIB
	PERLLIB
	PERLIO_DEBUG 
	JAVA_TOOL_OPTIONS
	SHELLOPTS
	GLOBIGNORE
	PS4
	BASH_ENV
	ENV
	TERMCAP
	TERMPATH
	TERMINFO_DIRS
	TERMINFO
	_RLD*
	LD_*
	PATH_LOCALE
	NLSPATH
	HOSTALIASES
	RES_OPTIONS
	LOCALDOMAIN
	CDPATH
	IFS
Environment variables to preserve:
	JAVA_HOME
	TRAVIS
	CI
	DEBIAN_FRONTEND
	XAUTHORIZATION
	XAUTHORITY
	PS2
	PS1
	PATH
	LS_COLORS
	KRB5CCNAME
	HOSTNAME
	HOME
	DISPLAY
	COLORS
Locale to use while parsing sudoers: C
Directory in which to store input/output logs: /var/log/sudo-io
File in which to store the input/output log: %{seq}
Add an entry to the utmp/utmpx file when allocating a pty
PAM service name to use
PAM service name to use for login shells
Create a new PAM session for the command to run in
Maximum I/O log sequence number: 0
Local IP address and netmask pairs:
	10.240.0.28/255.255.255.255
	172.17.0.1/255.255.0.0
Sudoers I/O plugin version 1.8.9p5
gzip version
gzip 1.6
Copyright (C) 2007, 2010, 2011 Free Software Foundation, Inc.
Copyright (C) 1993 Jean-loup Gailly.
This is free software.  You may redistribute copies of it under the terms of
the GNU General Public License <http://www.gnu.org/licenses/gpl.html>.
There is NO WARRANTY, to the extent permitted by law.
Written by Jean-loup Gailly.
zip version
Copyright (c) 1990-2008 Info-ZIP - Type 'zip "-L"' for software license.
This is Zip 3.0 (July 5th 2008), by Info-ZIP.
Currently maintained by E. Gordon.  Please send bug reports to
the authors using the web page at www.info-zip.org; see README for details.
Latest sources and executables are at ftp://ftp.info-zip.org/pub/infozip,
as of above date; see http://www.info-zip.org/ for other sites.
Compiled with gcc 4.8.2 for Unix (Linux ELF) on Oct 21 2013.
Zip special compilation options:
	USE_EF_UT_TIME       (store Universal Time)
	BZIP2_SUPPORT        (bzip2 library version 1.0.6, 6-Sept-2010)
	    bzip2 code and library copyright (c) Julian R Seward
	    (See the bzip2 license for terms of use)
	SYMLINK_SUPPORT      (symbolic links supported)
	LARGE_FILE_SUPPORT   (can read and write large files on file system)
	ZIP64_SUPPORT        (use Zip64 to store large files in archives)
	UNICODE_SUPPORT      (store and read UTF-8 Unicode paths)
	STORE_UNIX_UIDs_GIDs (store UID/GID sizes/values using new extra field)
	UIDGID_NOT_16BIT     (old Unix 16-bit UID/GID extra field not used)
	[encryption, version 2.91 of 05 Jan 2007] (modified for Zip 3)
Encryption notice:
	The encryption code of this program is not copyrighted and is
	put in the public domain.  It was originally written in Europe
	and, to the best of our knowledge, can be freely distributed
	in both source and object forms from any country, including
	the USA under License Exception TSU of the U.S. Export
	Administration Regulations (section 740.13(e)) of 6 June 2002.
Zip environment options:
             ZIP:  [none]
          ZIPOPT:  [none]
vim version
VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Nov 24 2016 16:43:18)
Included patches: 1-52
Extra patches: 8.0.0056
Modified by pkg-vim-maintainers@lists.alioth.debian.org
Compiled by buildd@
Huge version without GUI.  Features included (+) or not (-):
+acl             +farsi           +mouse_netterm   +syntax
+arabic          +file_in_path    +mouse_sgr       +tag_binary
+autocmd         +find_in_path    -mouse_sysmouse  +tag_old_static
-balloon_eval    +float           +mouse_urxvt     -tag_any_white
-browse          +folding         +mouse_xterm     -tcl
++builtin_terms  -footer          +multi_byte      +terminfo
+byte_offset     +fork()          +multi_lang      +termresponse
+cindent         +gettext         -mzscheme        +textobjects
-clientserver    -hangul_input    +netbeans_intg   +title
-clipboard       +iconv           +path_extra      -toolbar
+cmdline_compl   +insert_expand   -perl            +user_commands
+cmdline_hist    +jumplist        +persistent_undo +vertsplit
+cmdline_info    +keymap          +postscript      +virtualedit
+comments        +langmap         +printer         +visual
+conceal         +libcall         +profile         +visualextra
+cryptv          +linebreak       +python          +viminfo
+cscope          +lispindent      -python3         +vreplace
+cursorbind      +listcmds        +quickfix        +wildignore
+cursorshape     +localmap        +reltime         +wildmenu
+dialog_con      -lua             +rightleft       +windows
+diff            +menu            -ruby            +writebackup
+digraphs        +mksession       +scrollbind      -X11
-dnd             +modify_fname    +signs           -xfontset
-ebcdic          +mouse           +smartindent     -xim
+emacs_tags      -mouseshape      -sniff           -xsmp
+eval            +mouse_dec       +startuptime     -xterm_clipboard
+ex_extra        +mouse_gpm       +statusline      -xterm_save
+extra_search    -mouse_jsbterm   -sun_workshop    -xpm
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  fall-back for $VIM: "/usr/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H     -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1      
Linking: gcc   -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,--as-needed -o vim        -lm -ltinfo -lnsl  -lselinux  -lacl -lattr -lgpm -ldl    -L/usr/lib/python2.7/config-x86_64-linux-gnu -lpython2.7 -lpthread -ldl -lutil -lm -Xlinker -export-dynamic -Wl,-O1 -Wl,-Bsymbolic-functions      
iptables version
iptables v1.4.21
curl version
curl 7.35.0 (x86_64-pc-linux-gnu) libcurl/7.35.0 OpenSSL/1.0.1f zlib/1.2.8 libidn/1.28 librtmp/2.3
wget version
GNU Wget 1.15 built on linux-gnu.
rsync version
rsync  version 3.1.0  protocol version 31
gimme version
v1.2.0
nvm version
0.33.6
perlbrew version
/home/travis/perl5/perlbrew/bin/perlbrew  - App::perlbrew/0.80
phpenv version
rbenv 1.1.1-25-g6aa70b6
rvm version
rvm 1.29.3 (latest) by Michal Papis, Piotr Kuczynski, Wayne E. Seguin [https://rvm.io]
default ruby version
ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-linux]
CouchDB version
couchdb 1.6.1
ElasticSearch version
5.5.0
Installed Firefox version
firefox 56.0.2
MongoDB version
MongoDB 3.4.10
PhantomJS version
2.1.1
Pre-installed PostgreSQL versions
9.2.24
9.3.20
9.4.15
9.5.10
9.6.6
RabbitMQ Version
3.6.14
Redis version
redis-server 4.0.6
riak version
2.2.3
Pre-installed Go versions
1.7.4
ant version
Apache Ant(TM) version 1.9.3 compiled on April 8 2014
mvn version
Apache Maven 3.5.2 (138edd61fd100ec658bfa2d307c43b76940a5d7d; 2017-10-18T07:58:13Z)
Maven home: /usr/local/maven-3.5.2
Java version: 1.8.0_151, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-8-oracle/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.4.0-98-generic", arch: "amd64", family: "unix"
gradle version
------------------------------------------------------------
Gradle 4.0.1
------------------------------------------------------------
Build time:   2017-07-07 14:02:41 UTC
Revision:     38e5dc0f772daecca1d2681885d3d85414eb6826
Groovy:       2.4.11
Ant:          Apache Ant(TM) version 1.9.6 compiled on June 29 2015
JVM:          1.8.0_151 (Oracle Corporation 25.151-b12)
OS:           Linux 4.4.0-98-generic amd64
lein version
Leiningen 2.8.1 on Java 1.8.0_151 Java HotSpot(TM) 64-Bit Server VM
Pre-installed Node.js versions
v4.8.6
v6.12.0
v6.12.1
v8.9
v8.9.1
phpenv versions
  system
  5.6
* 5.6.32 (set by /home/travis/.phpenv/version)
  7.0
  7.0.25
  7.1
  7.1.11
  hhvm
  hhvm-stable
composer --version
Composer version 1.5.2 2017-09-11 16:59:25
Pre-installed Ruby versions
ruby-2.2.7
ruby-2.3.4
ruby-2.4.1
removed ‘/etc/apt/sources.list.d/basho_riak.list’
Executing: /tmp/tmp.qLWGriNsXE/gpg.1.sh --keyserver
hkp://keyserver.ubuntu.com:80
--recv
EA312927
gpg: requesting key EA312927 from hkp server keyserver.ubuntu.com
gpg: key EA312927: public key "MongoDB 3.2 Release Signing Key <packaging@mongodb.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1  (RSA: 1)
W: The repository 'http://ppa.launchpad.net/couchdb/stable/ubuntu trusty Release' does not have a Release file.
W: Failed to fetch https://packagecloud.io/computology/apt-backport/ubuntu/dists/trusty/InRelease  Failed to connect to packagecloud.io port 443: Connection timed out
W: Failed to fetch http://repo.mongodb.org/apt/ubuntu/dists/trusty/mongodb-org/3.4/InRelease  Could not connect to repo.mongodb.org:80 (52.222.215.211), connection timed out
W: Failed to fetch http://ppa.launchpad.net/openjdk-r/ppa/ubuntu/dists/trusty/InRelease  Could not connect to ppa.launchpad.net:80 (91.189.95.83), connection timed out
W: Failed to fetch http://ppa.launchpad.net/pollinate/ppa/ubuntu/dists/trusty/InRelease  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/webupd8team/java/ubuntu/dists/trusty/InRelease  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/chris-lea/redis-server/ubuntu/dists/trusty/main/binary-amd64/Packages  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/chris-lea/redis-server/ubuntu/dists/trusty/main/binary-i386/Packages  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/chris-lea/redis-server/ubuntu/dists/trusty/main/i18n/Translation-en  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/git-core/ppa/ubuntu/dists/trusty/main/binary-amd64/Packages  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/git-core/ppa/ubuntu/dists/trusty/main/binary-i386/Packages  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/git-core/ppa/ubuntu/dists/trusty/main/i18n/Translation-en  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/couchdb/stable/ubuntu/dists/trusty/main/binary-amd64/Packages  Unable to connect to ppa.launchpad.net:http:
W: Failed to fetch http://ppa.launchpad.net/couchdb/stable/ubuntu/dists/trusty/main/binary-i386/Packages  Unable to connect to ppa.launchpad.net:http:
W: Some index files failed to download. They have been ignored, or old ones used instead.
127.0.0.1 localhost nettuno travis vagrant
127.0.1.1 travis-job-3132ee34-8996-4436-ab25-1b97c4f0caa0 travis-job-3132ee34-8996-4436-ab25-1b97c4f0caa0 ip4-loopback trusty64
git.checkout
1.61s
$ git clone --depth=50 https://github.com/jerryscript-project/jerryscript.git jerryscript-project/jerryscript
Cloning into 'jerryscript-project/jerryscript'...
remote: Counting objects: 3355, done.
remote: Compressing objects: 100% (1518/1518), done.
remote: Total 3355 (delta 2413), reused 2275 (delta 1754), pack-reused 0
Receiving objects: 100% (3355/3355), 2.55 MiB | 8.58 MiB/s, done.
Resolving deltas: 100% (2413/2413), done.
$ cd jerryscript-project/jerryscript
0.83s
$ git fetch origin +refs/pull/2133/merge:
remote: Counting objects: 17, done.
remote: Compressing objects: 100% (6/6), done.
remote: Total 17 (delta 12), reused 14 (delta 10), pack-reused 0
Unpacking objects: 100% (17/17), done.
From https://github.com/jerryscript-project/jerryscript
 * branch            refs/pull/2133/merge -> FETCH_HEAD
$ git checkout -qf FETCH_HEAD
apt
Adding APT Sources (BETA)
$ export DEBIAN_FRONTEND=noninteractive
1.49s
$ sudo -E apt-add-repository -y "ppa:team-gcc-arm-embedded/ppa"
gpg: keyring `/tmp/tmp06zwp5cm/secring.gpg' created
gpg: keyring `/tmp/tmp06zwp5cm/pubring.gpg' created
gpg: requesting key F64D33B0 from hkp server keyserver.ubuntu.com
gpg: /tmp/tmp06zwp5cm/trustdb.gpg: trustdb created
gpg: key F64D33B0: public key "Launchpad PPA for GCC ARM Embedded Maintainers" imported
gpg: no ultimately trusted keys found
gpg: Total number processed: 1
gpg:               imported: 1  (RSA: 1)
OK
Installing APT Packages (BETA)
$ export DEBIAN_FRONTEND=noninteractive
8.69s
$ sudo -E apt-get -yq update &>> ~/apt-get-update.log
93.61s
$ sudo -E apt-get -yq --no-install-suggests --no-install-recommends --force-yes install gcc-arm-embedded
Reading package lists...
Building dependency tree...
Reading state information...
The following NEW packages will be installed:
  gcc-arm-embedded
0 upgraded, 1 newly installed, 0 to remove and 15 not upgraded.
Need to get 65.0 MB of archives.
After this operation, 434 MB of additional disk space will be used.
Get:1 http://ppa.launchpad.net/team-gcc-arm-embedded/ppa/ubuntu trusty/main amd64 gcc-arm-embedded amd64 6-2017q2-2~trusty1 [65.0 MB]
Fetched 65.0 MB in 1min 25s (760 kB/s)
Selecting previously unselected package gcc-arm-embedded.
(Reading database ... 93696 files and directories currently installed.)
Preparing to unpack .../gcc-arm-embedded_6-2017q2-2~trusty1_amd64.deb ...
Unpacking gcc-arm-embedded (6-2017q2-2~trusty1) ...
Processing triggers for ccache (3.1.9-1) ...
Updating symlinks in /usr/lib/ccache ...
Setting up gcc-arm-embedded (6-2017q2-2~trusty1) ...
W: --force-yes is deprecated, use one of the options starting with --allow instead.
Setting environment variables from .travis.yml
$ export JOBNAME="Mbed/K64F Build Test"
$ export CC=gcc
$ gcc --version
gcc (Ubuntu 4.8.4-2ubuntu1~14.04.3) 4.8.4
Copyright (C) 2013 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
before_install
9.29s
$ tools/apt-get-install-deps.sh
Ign:1 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty InRelease
Hit:2 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates InRelease
Hit:3 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-backports InRelease
Hit:4 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty Release
Hit:5 http://apt.postgresql.org/pub/repos/apt trusty-pgdg InRelease
Ign:6 http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.4 InRelease
Hit:8 http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.4 Release
Ign:10 http://dl.google.com/linux/chrome/deb stable InRelease
Ign:7 http://toolbelt.heroku.com/ubuntu ./ InRelease
Hit:11 http://dl.google.com/linux/chrome/deb stable Release
Get:9 http://dl.bintray.com/apache/cassandra 39x InRelease [3,168 B]
Hit:13 http://ppa.launchpad.net/chris-lea/redis-server/ubuntu trusty InRelease
Hit:14 http://security.ubuntu.com/ubuntu trusty-security InRelease
Hit:12 http://toolbelt.heroku.com/ubuntu ./ Release
Hit:16 https://dl.hhvm.com/ubuntu trusty InRelease
Hit:17 https://download.docker.com/linux/ubuntu trusty InRelease
Ign:18 http://ppa.launchpad.net/couchdb/stable/ubuntu trusty InRelease
Hit:21 http://ppa.launchpad.net/git-core/ppa/ubuntu trusty InRelease
Hit:23 http://ppa.launchpad.net/openjdk-r/ppa/ubuntu trusty InRelease
Hit:24 https://packagecloud.io/computology/apt-backport/ubuntu trusty InRelease
Hit:25 http://ppa.launchpad.net/pollinate/ppa/ubuntu trusty InRelease
Hit:26 https://packagecloud.io/github/git-lfs/ubuntu trusty InRelease
Hit:27 https://packagecloud.io/rabbitmq/rabbitmq-server/ubuntu trusty InRelease
Hit:28 http://ppa.launchpad.net/team-gcc-arm-embedded/ppa/ubuntu trusty InRelease
Hit:29 http://ppa.launchpad.net/webupd8team/java/ubuntu trusty InRelease
Hit:30 http://ppa.launchpad.net/couchdb/stable/ubuntu trusty Release
Fetched 3,168 B in 1s (1,915 B/s)
Reading package lists...
W: http://ppa.launchpad.net/couchdb/stable/ubuntu/dists/trusty/Release.gpg: Signature by key 15866BAFD9BCC4F3C1E0DFC7D69548E1C17EAB57 uses weak digest algorithm (SHA1)
Reading package lists...
Building dependency tree...
Reading state information...
gcc is already the newest version (4:4.8.2-1ubuntu6).
gcc set to manually installed.
make is already the newest version (3.81-8.2ubuntu3).
make set to manually installed.
python is already the newest version (2.7.5-5ubuntu3).
The following additional packages will be installed:
  cmake-data gcc-4.8-multilib lib32asan0 lib32atomic1 lib32gcc-4.8-dev
  lib32gcc1 lib32gomp1 lib32itm1 lib32quadmath0 libarchive13
  libboost-program-options1.54.0 libboost-system1.54.0 libboost-thread1.54.0
  libboost-wave1.54.0 libc6-dev-i386 libc6-dev-x32 libc6-i386 libc6-x32
  libnettle4 libtinyxml2-0.0.0 libx32asan0 libx32atomic1 libx32gcc-4.8-dev
  libx32gcc1 libx32gomp1 libx32itm1 libx32quadmath0
Suggested packages:
  codeblocks eclipse doxygen-latex doxygen-doc doxygen-gui graphviz lrzip
The following NEW packages will be installed:
  cmake cmake-data cppcheck doxygen gcc-4.8-multilib gcc-multilib lib32asan0
  lib32atomic1 lib32gcc-4.8-dev lib32gcc1 lib32gomp1 lib32itm1 lib32quadmath0
  libarchive13 libboost-program-options1.54.0 libboost-system1.54.0
  libboost-thread1.54.0 libboost-wave1.54.0 libc6-dev-i386 libc6-dev-x32
  libc6-i386 libc6-x32 libnettle4 libtinyxml2-0.0.0 libx32asan0 libx32atomic1
  libx32gcc-4.8-dev libx32gcc1 libx32gomp1 libx32itm1 libx32quadmath0 vera++
0 upgraded, 32 newly installed, 0 to remove and 15 not upgraded.
Need to get 18.3 MB of archives.
After this operation, 77.7 MB of additional disk space will be used.
Get:1 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 libc6-i386 amd64 2.19-0ubuntu6.13 [2,207 kB]
Get:2 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 libc6-dev-i386 amd64 2.19-0ubuntu6.13 [1,152 kB]
Get:3 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 libc6-x32 amd64 2.19-0ubuntu6.13 [2,433 kB]
Get:4 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 libc6-dev-x32 amd64 2.19-0ubuntu6.13 [1,414 kB]
Get:5 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 lib32gcc1 amd64 1:4.9.3-0ubuntu4 [47.8 kB]
Get:6 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 libx32gcc1 amd64 1:4.9.3-0ubuntu4 [38.6 kB]
Get:7 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 lib32gomp1 amd64 4.8.4-2ubuntu1~14.04.3 [25.0 kB]
Get:8 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 libx32gomp1 amd64 4.8.4-2ubuntu1~14.04.3 [22.5 kB]
Get:9 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 lib32itm1 amd64 4.8.4-2ubuntu1~14.04.3 [28.5 kB]
Get:10 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 libx32itm1 amd64 4.8.4-2ubuntu1~14.04.3 [28.4 kB]
Get:11 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 lib32atomic1 amd64 4.8.4-2ubuntu1~14.04.3 [8,248 B]
Get:12 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 libx32atomic1 amd64 4.8.4-2ubuntu1~14.04.3 [8,602 B]
Get:13 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 lib32asan0 amd64 4.8.4-2ubuntu1~14.04.3 [64.2 kB]
Get:14 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 libx32asan0 amd64 4.8.4-2ubuntu1~14.04.3 [63.0 kB]
Get:15 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 lib32quadmath0 amd64 4.8.4-2ubuntu1~14.04.3 [187 kB]
Get:16 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 libx32quadmath0 amd64 4.8.4-2ubuntu1~14.04.3 [128 kB]
Get:17 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 lib32gcc-4.8-dev amd64 4.8.4-2ubuntu1~14.04.3 [1,686 kB]
Get:18 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 libx32gcc-4.8-dev amd64 4.8.4-2ubuntu1~14.04.3 [1,557 kB]
Get:19 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 gcc-4.8-multilib amd64 4.8.4-2ubuntu1~14.04.3 [978 B]
Get:20 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/main amd64 gcc-multilib amd64 4:4.8.2-1ubuntu6 [1,024 B]
Get:21 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 libnettle4 amd64 2.7.1-1ubuntu0.2 [102 kB]
Get:22 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 libarchive13 amd64 3.1.2-7ubuntu2.4 [262 kB]
Get:23 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 libboost-program-options1.54.0 amd64 1.54.0-4ubuntu3.1 [115 kB]
Get:24 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 libboost-system1.54.0 amd64 1.54.0-4ubuntu3.1 [10.1 kB]
Get:25 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 libboost-thread1.54.0 amd64 1.54.0-4ubuntu3.1 [26.5 kB]
Get:26 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/universe amd64 libboost-wave1.54.0 amd64 1.54.0-4ubuntu3.1 [161 kB]
Get:27 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/universe amd64 libtinyxml2-0.0.0 amd64 0~git20120518.1.a2ae54e-1 [27.4 kB]
Get:28 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/main amd64 cmake-data all 2.8.12.2-0ubuntu3 [676 kB]
Get:29 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/main amd64 cmake amd64 2.8.12.2-0ubuntu3 [2,618 kB]
Get:30 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/universe amd64 cppcheck amd64 1.61-1 [684 kB]
Get:31 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/main amd64 doxygen amd64 1.8.6-2 [2,408 kB]
Get:32 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty/universe amd64 vera++ amd64 1.2.1-2 [136 kB]
Fetched 18.3 MB in 0s (43.8 MB/s)
Extracting templates from packages: 100%
Selecting previously unselected package libc6-i386.
(Reading database ... 99615 files and directories currently installed.)
Preparing to unpack .../libc6-i386_2.19-0ubuntu6.13_amd64.deb ...
Unpacking libc6-i386 (2.19-0ubuntu6.13) ...
Selecting previously unselected package libc6-dev-i386.
Preparing to unpack .../libc6-dev-i386_2.19-0ubuntu6.13_amd64.deb ...
Unpacking libc6-dev-i386 (2.19-0ubuntu6.13) ...
Selecting previously unselected package libc6-x32.
Preparing to unpack .../libc6-x32_2.19-0ubuntu6.13_amd64.deb ...
Unpacking libc6-x32 (2.19-0ubuntu6.13) ...
Selecting previously unselected package libc6-dev-x32.
Preparing to unpack .../libc6-dev-x32_2.19-0ubuntu6.13_amd64.deb ...
Unpacking libc6-dev-x32 (2.19-0ubuntu6.13) ...
Selecting previously unselected package lib32gcc1.
Preparing to unpack .../lib32gcc1_1%3a4.9.3-0ubuntu4_amd64.deb ...
Unpacking lib32gcc1 (1:4.9.3-0ubuntu4) ...
Selecting previously unselected package libx32gcc1.
Preparing to unpack .../libx32gcc1_1%3a4.9.3-0ubuntu4_amd64.deb ...
Unpacking libx32gcc1 (1:4.9.3-0ubuntu4) ...
Selecting previously unselected package lib32gomp1.
Preparing to unpack .../lib32gomp1_4.8.4-2ubuntu1~14.04.3_amd64.deb ...
Unpacking lib32gomp1 (4.8.4-2ubuntu1~14.04.3) ...
Selecting previously unselected package libx32gomp1.
Preparing to unpack .../libx32gomp1_4.8.4-2ubuntu1~14.04.3_amd64.deb ...
Unpacking libx32gomp1 (4.8.4-2ubuntu1~14.04.3) ...
Selecting previously unselected package lib32itm1.
Preparing to unpack .../lib32itm1_4.8.4-2ubuntu1~14.04.3_amd64.deb ...
Unpacking lib32itm1 (4.8.4-2ubuntu1~14.04.3) ...
Selecting previously unselected package libx32itm1.
Preparing to unpack .../libx32itm1_4.8.4-2ubuntu1~14.04.3_amd64.deb ...
Unpacking libx32itm1 (4.8.4-2ubuntu1~14.04.3) ...
Selecting previously unselected package lib32atomic1.
Preparing to unpack .../lib32atomic1_4.8.4-2ubuntu1~14.04.3_amd64.deb ...
Unpacking lib32atomic1 (4.8.4-2ubuntu1~14.04.3) ...
Selecting previously unselected package libx32atomic1.
Preparing to unpack .../libx32atomic1_4.8.4-2ubuntu1~14.04.3_amd64.deb ...
Unpacking libx32atomic1 (4.8.4-2ubuntu1~14.04.3) ...
Selecting previously unselected package lib32asan0.
Preparing to unpack .../lib32asan0_4.8.4-2ubuntu1~14.04.3_amd64.deb ...
Unpacking lib32asan0 (4.8.4-2ubuntu1~14.04.3) ...
Selecting previously unselected package libx32asan0.
Preparing to unpack .../libx32asan0_4.8.4-2ubuntu1~14.04.3_amd64.deb ...
Unpacking libx32asan0 (4.8.4-2ubuntu1~14.04.3) ...
Selecting previously unselected package lib32quadmath0.
Preparing to unpack .../lib32quadmath0_4.8.4-2ubuntu1~14.04.3_amd64.deb ...
Unpacking lib32quadmath0 (4.8.4-2ubuntu1~14.04.3) ...
Selecting previously unselected package libx32quadmath0.
Preparing to unpack .../libx32quadmath0_4.8.4-2ubuntu1~14.04.3_amd64.deb ...
Unpacking libx32quadmath0 (4.8.4-2ubuntu1~14.04.3) ...
Selecting previously unselected package lib32gcc-4.8-dev.
Preparing to unpack .../lib32gcc-4.8-dev_4.8.4-2ubuntu1~14.04.3_amd64.deb ...
Unpacking lib32gcc-4.8-dev (4.8.4-2ubuntu1~14.04.3) ...
Selecting previously unselected package libx32gcc-4.8-dev.
Preparing to unpack .../libx32gcc-4.8-dev_4.8.4-2ubuntu1~14.04.3_amd64.deb ...
Unpacking libx32gcc-4.8-dev (4.8.4-2ubuntu1~14.04.3) ...
Selecting previously unselected package gcc-4.8-multilib.
Preparing to unpack .../gcc-4.8-multilib_4.8.4-2ubuntu1~14.04.3_amd64.deb ...
Unpacking gcc-4.8-multilib (4.8.4-2ubuntu1~14.04.3) ...
Selecting previously unselected package gcc-multilib.
Preparing to unpack .../gcc-multilib_4%3a4.8.2-1ubuntu6_amd64.deb ...
Unpacking gcc-multilib (4:4.8.2-1ubuntu6) ...
Selecting previously unselected package libnettle4:amd64.
Preparing to unpack .../libnettle4_2.7.1-1ubuntu0.2_amd64.deb ...
Unpacking libnettle4:amd64 (2.7.1-1ubuntu0.2) ...
Selecting previously unselected package libarchive13:amd64.
Preparing to unpack .../libarchive13_3.1.2-7ubuntu2.4_amd64.deb ...
Unpacking libarchive13:amd64 (3.1.2-7ubuntu2.4) ...
Selecting previously unselected package libboost-program-options1.54.0:amd64.
Preparing to unpack .../libboost-program-options1.54.0_1.54.0-4ubuntu3.1_amd64.deb ...
Unpacking libboost-program-options1.54.0:amd64 (1.54.0-4ubuntu3.1) ...
Selecting previously unselected package libboost-system1.54.0:amd64.
Preparing to unpack .../libboost-system1.54.0_1.54.0-4ubuntu3.1_amd64.deb ...
Unpacking libboost-system1.54.0:amd64 (1.54.0-4ubuntu3.1) ...
Selecting previously unselected package libboost-thread1.54.0:amd64.
Preparing to unpack .../libboost-thread1.54.0_1.54.0-4ubuntu3.1_amd64.deb ...
Unpacking libboost-thread1.54.0:amd64 (1.54.0-4ubuntu3.1) ...
Selecting previously unselected package libboost-wave1.54.0:amd64.
Preparing to unpack .../libboost-wave1.54.0_1.54.0-4ubuntu3.1_amd64.deb ...
Unpacking libboost-wave1.54.0:amd64 (1.54.0-4ubuntu3.1) ...
Selecting previously unselected package libtinyxml2-0.0.0:amd64.
Preparing to unpack .../libtinyxml2-0.0.0_0~git20120518.1.a2ae54e-1_amd64.deb ...
Unpacking libtinyxml2-0.0.0:amd64 (0~git20120518.1.a2ae54e-1) ...
Selecting previously unselected package cmake-data.
Preparing to unpack .../cmake-data_2.8.12.2-0ubuntu3_all.deb ...
ERROR: cmake-data is broken - called emacs-package-install as a new-style add-on, but has no compat file.
Unpacking cmake-data (2.8.12.2-0ubuntu3) ...
Selecting previously unselected package cmake.
Preparing to unpack .../cmake_2.8.12.2-0ubuntu3_amd64.deb ...
Unpacking cmake (2.8.12.2-0ubuntu3) ...
Selecting previously unselected package cppcheck.
Preparing to unpack .../cppcheck_1.61-1_amd64.deb ...
Unpacking cppcheck (1.61-1) ...
Selecting previously unselected package doxygen.
Preparing to unpack .../doxygen_1.8.6-2_amd64.deb ...
Unpacking doxygen (1.8.6-2) ...
Selecting previously unselected package vera++.
Preparing to unpack .../vera++_1.2.1-2_amd64.deb ...
Unpacking vera++ (1.2.1-2) ...
Processing triggers for ccache (3.1.9-1) ...
Updating symlinks in /usr/lib/ccache ...
Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
Setting up libc6-i386 (2.19-0ubuntu6.13) ...
Setting up libc6-dev-i386 (2.19-0ubuntu6.13) ...
Setting up libc6-x32 (2.19-0ubuntu6.13) ...
Setting up libc6-dev-x32 (2.19-0ubuntu6.13) ...
Setting up lib32gcc1 (1:4.9.3-0ubuntu4) ...
Setting up libx32gcc1 (1:4.9.3-0ubuntu4) ...
Setting up lib32gomp1 (4.8.4-2ubuntu1~14.04.3) ...
Setting up libx32gomp1 (4.8.4-2ubuntu1~14.04.3) ...
Setting up lib32itm1 (4.8.4-2ubuntu1~14.04.3) ...
Setting up libx32itm1 (4.8.4-2ubuntu1~14.04.3) ...
Setting up lib32atomic1 (4.8.4-2ubuntu1~14.04.3) ...
Setting up libx32atomic1 (4.8.4-2ubuntu1~14.04.3) ...
Setting up lib32asan0 (4.8.4-2ubuntu1~14.04.3) ...
Setting up libx32asan0 (4.8.4-2ubuntu1~14.04.3) ...
Setting up lib32quadmath0 (4.8.4-2ubuntu1~14.04.3) ...
Setting up libx32quadmath0 (4.8.4-2ubuntu1~14.04.3) ...
Setting up lib32gcc-4.8-dev (4.8.4-2ubuntu1~14.04.3) ...
Setting up libx32gcc-4.8-dev (4.8.4-2ubuntu1~14.04.3) ...
Setting up gcc-4.8-multilib (4.8.4-2ubuntu1~14.04.3) ...
Setting up gcc-multilib (4:4.8.2-1ubuntu6) ...
Setting up libnettle4:amd64 (2.7.1-1ubuntu0.2) ...
Setting up libarchive13:amd64 (3.1.2-7ubuntu2.4) ...
Setting up libboost-program-options1.54.0:amd64 (1.54.0-4ubuntu3.1) ...
Setting up libboost-system1.54.0:amd64 (1.54.0-4ubuntu3.1) ...
Setting up libboost-thread1.54.0:amd64 (1.54.0-4ubuntu3.1) ...
Setting up libboost-wave1.54.0:amd64 (1.54.0-4ubuntu3.1) ...
Setting up libtinyxml2-0.0.0:amd64 (0~git20120518.1.a2ae54e-1) ...
Setting up cmake-data (2.8.12.2-0ubuntu3) ...
Install cmake-data for emacs23
install/cmake-data: Byte-compiling for emacs23
Wrote /usr/share/emacs23/site-lisp/cmake-data/cmake-mode.elc
Install cmake-data for emacs24
install/cmake-data: Byte-compiling for emacs24
Wrote /usr/share/emacs24/site-lisp/cmake-data/cmake-mode.elc
Setting up cmake (2.8.12.2-0ubuntu3) ...
Setting up cppcheck (1.61-1) ...
Setting up doxygen (1.8.6-2) ...
Setting up vera++ (1.2.1-2) ...
Processing triggers for libc-bin (2.19-0ubuntu6.13) ...
1.26s
$ make -f ./targets/mbed/Makefile.travis install
sudo apt-get install -q -y ninja-build libffi-dev libssl-dev
Reading package lists...
Building dependency tree...
Reading state information...
libffi-dev is already the newest version (3.1~rc1+r3.0.13-12ubuntu0.2).
libssl-dev is already the newest version (1.0.1f-1ubuntu2.23).
libssl-dev set to manually installed.
The following NEW packages will be installed:
  ninja-build
0 upgraded, 1 newly installed, 0 to remove and 15 not upgraded.
Need to get 79.6 kB of archives.
After this operation, 317 kB of additional disk space will be used.
Get:1 http://us-central1.gce.archive.ubuntu.com/ubuntu trusty-updates/main amd64 ninja-build amd64 1.3.4-1.1ubuntu0.14.04.1 [79.6 kB]
Fetched 79.6 kB in 0s (4,697 kB/s)
Selecting previously unselected package ninja-build.
(Reading database ... 101322 files and directories currently installed.)
Preparing to unpack .../ninja-build_1.3.4-1.1ubuntu0.14.04.1_amd64.deb ...
Unpacking ninja-build (1.3.4-1.1ubuntu0.14.04.1) ...
Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
Setting up ninja-build (1.3.4-1.1ubuntu0.14.04.1) ...
pyenv global 2.7.13 # force the python version to a newer one
pyenv: version `2.7.13' not installed
make: *** [install-yotta] Error 1
The command "make -f ./targets/mbed/Makefile.travis install" failed and exited with 2 during .
Your build has been stopped.
```

jiangzidong commented:
@LaszloLango Could you review this one again since it is updated. Thanks

zherczeg commented:
Only mbed fails, but that will be green on master.
